tests/utils/test_cycle_consistency.py

```diff
@@ -0,0 +1,299 @@
+        (1, 2): Rot3(),
+        (2, 3): Rot3(),
+        (1, 3): Rot3(),
+        (3, 4): Rot3(),
+        (1, 5): Rot3(),
+        (3, 5): Rot3(),
+    }
+
+    for extraction_fn in [cycle_utils.extract_triplets_adjacency_list_intersection, extract_triplets_brute_force]:
+
+        triplets = extraction_fn(i2Ri1_dict)
+        assert len(triplets) == 2
+        assert triplets[0] == (1, 2, 3)
+        assert triplets[1] == (1, 3, 5)
+
+        assert isinstance(triplets, list)
+
+
+def test_compute_cycle_error_known_GT() -> None:
+    """Ensure cycle error is computed correctly within a triplet, when ground truth is known.
+
+    Imagine 3 poses, all centered at the origin, at different orientations.
+
+    Ground truth poses:
+       Let i0 face along +x axis (0 degrees in yaw)
+       Let i2 have a 30 degree rotation from the +x axis.
+       Let i4 have a 90 degree rotation from the +x axis.
+
+    However, suppose one edge measurement is corrupted (from i0 -> i4) by 5 degrees.
+    """
+    i2Ri0 = Rot3(Rotation.from_euler("y", 30, degrees=True).as_matrix())
+    i4Ri2 = Rot3(Rotation.from_euler("y", 60, degrees=True).as_matrix())
+    i4Ri0 = Rot3(Rotation.from_euler("y", 95, degrees=True).as_matrix())
+
+    cycle_nodes = [0, 2, 4]
+    i2Ri1_dict = {
+        (0, 2): i2Ri0,  # edge i
+        (2, 4): i4Ri2,  # edge j
+        (0, 4): i4Ri0,  # edge k
+    }
+
+    two_view_reports_dict = {}
+    # rest of attributes will default to None
+    two_view_reports_dict[(0, 4)] = TwoViewEstimationReport(
+        v_corr_idxs=np.array([]),  # dummy array
+        num_inliers_est_model=10,  # dummy value
+        R_error_deg=5,
+        U_error_deg=0,
+    )
+
+    two_view_reports_dict[(0, 2)] = TwoViewEstimationReport(
+        v_corr_idxs=np.array([]),  # dummy array
+        num_inliers_est_model=10,  # dummy value
+        R_error_deg=0,
+        U_error_deg=0,
+    )
+
+    two_view_reports_dict[(2, 4)] = TwoViewEstimationReport(
+        v_corr_idxs=np.array([]),  # dummy array
+        num_inliers_est_model=10,  # dummy value
+        R_error_deg=0,
+        U_error_deg=0,
+    )
+
+    cycle_error, max_rot_error, max_trans_error = cycle_utils.compute_cycle_error(
+        i2Ri1_dict, cycle_nodes, two_view_reports_dict
+    )
+
+    assert np.isclose(cycle_error, 5)
+    assert np.isclose(max_rot_error, 5)
+    assert max_trans_error == 0
+
+
+def test_compute_cycle_error_unknown_GT() -> None:
+    """Ensure cycle error is computed correctly within a triplet, when ground truth is known.
+
+    Imagine 3 poses, all centered at the origin, at different orientations.
+
+    Ground truth poses:
+       Let i0 face along +x axis (0 degrees in yaw)
+       Let i2 have a 30 degree rotation from the +x axis.
+       Let i4 have a 90 degree rotation from the +x axis.
+
+    However, suppose one edge measurement is corrupted (from i0 -> i4) by 5 degrees.
+    """
+    i2Ri0 = Rot3(Rotation.from_euler("y", 30, degrees=True).as_matrix())
```

Comment:
Same, gtsam doesnt need a matrix to construct a rotation.

Reply:
I updated it now to use `Rot3.Ry(np.deg2rad(angle_deg))` here